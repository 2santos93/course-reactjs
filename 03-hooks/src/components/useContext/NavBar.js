import { useContext } from "react";
import { NavLink } from "react-router-dom";

export const NavBar = () => {

  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <NavLink className="navbar-brand" to="/">
          useContext
        </NavLink>
        <div className="collapse navbar-collapse">
          
          <div className="navbar-nav">
            <NavLink activeClassName="active" className="nav-link" exact to="/">Home</NavLink>
            <NavLink activeClassName="active" className="nav-link" to="/login">Login</NavLink>
            <NavLink activeClassName="active" className="nav-link" to="/about">About</NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};
