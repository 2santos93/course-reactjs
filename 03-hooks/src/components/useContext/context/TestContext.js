import { createContext } from "react";

export const TestContext = createContext({msg: 'no tengo provider :('});