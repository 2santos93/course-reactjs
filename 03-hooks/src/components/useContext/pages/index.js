import { Home } from "./Home"
import { Login } from "./Login"
import { About } from "./About"

export {
    Home,
    Login,
    About
}