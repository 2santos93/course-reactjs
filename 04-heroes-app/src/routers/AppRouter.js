import React, { useContext } from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";

import { AuthContext } from "../auth/AuthContext";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {
  const { state } = useContext(AuthContext);

  return (
    // <Router>
    //     <Switch>
    //       <Route exact path="/login" component={    (props) => (
    //         (!state.logged)
    //           ? (<LoginScreen {...props}/>)
    //           : (<Redirect to='/' />))
    //       }/>

    //       <Route path="/" component={ (props) => (
    //           (state.logged)
    //             ? (<DashboardRoutes {...props}/>)
    //             : (<Redirect to='/login' />)
    //         )
    //       }/>
    //     </Switch>
    // </Router>

    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
            isAuthenticated={state.logged}
          />

          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticated={state.logged}
          />
        </Switch>
      </div>
    </Router>
  );
};
