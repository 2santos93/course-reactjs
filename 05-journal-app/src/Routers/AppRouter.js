import React, { useEffect, useState } from 'react'
import { 
  Switch,
  BrowserRouter as Router,
  Redirect
  } from 'react-router-dom';
import { getAuth, onAuthStateChanged } from "firebase/auth";

import { JournalScreen } from '../Components/Journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

  const dispatch = useDispatch();
  const [state, setState] = useState({loading: true, isLogged: false})

  useEffect(() => {
    onAuthStateChanged(getAuth(), (user) => {
      if(user?.uid){
        dispatch(login(user.uid, user.displayName));
        setState((prev) => setState({...prev, isLogged:true}));
      }else{
        setState((prev) => setState({...prev, isLogged:false}));
      }
      
      setState((prev) => setState({...prev, loading:false}));
    });
  }, [setState, dispatch])
  
if(state?.loading){
  return <h1>Loading</h1>
}

  return (
    <Router>
      <Switch>
        <PublicRoute component={AuthRouter} path='/auth' isLogged={state.isLogged}/>

        <PrivateRoute exact path='/' component={JournalScreen} isLogged={state.isLogged}/>

        <Redirect to='/auth/login' />
      </Switch>
    </Router>
  )
}
