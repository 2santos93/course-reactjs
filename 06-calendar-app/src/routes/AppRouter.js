import React, { useEffect } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import { Redirect } from 'react-router';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { useDispatch, useSelector } from 'react-redux';
import { checkingStart } from '../actions/auth';
import { PrivateRoute } from '../components/router/PrivateRoute';
import { PublicRoute } from '../components/router/PublicRoute';

export const AppRouter = () => {
  const dispatch = useDispatch();
  const { user } = useSelector((state) => state.auth);

  useEffect(() => {
    dispatch(checkingStart());
  }, []);

  return (
    <Router>
      <Switch>
        <PublicRoute
          component={LoginScreen}
          path="/login"
          isLogged={!!user?.uid}
          exact
        />
        <PrivateRoute
          exact
          path="/"
          component={CalendarScreen}
          isLogged={!!user?.uid}
        />
        <Redirect to="/" />
      </Switch>
    </Router>
  );
};
