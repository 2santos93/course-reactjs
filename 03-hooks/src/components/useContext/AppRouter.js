import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { TestContext } from "./context/TestContext";
import { NavBar } from "./NavBar";

import { About, Home, Login } from "./pages";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

        <Switch>
          <Route exact path="/">
            <Home />
          </Route>

          <TestContext.Provider value={{msg:'con provider pa'}} >
            <Route path="/about">
              <About />
            </Route>
          </TestContext.Provider>
          
          <Route exact path="/login">
            <Login />
          </Route>

          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};
