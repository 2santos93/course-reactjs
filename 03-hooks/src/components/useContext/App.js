import { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './context/UserContext'

export const App = () => {

    const [user, setUser] = useState({});

    return (
        <UserContext.Provider value={{user, setUser}}>
            <AppRouter />
        </UserContext.Provider>
    )
}
