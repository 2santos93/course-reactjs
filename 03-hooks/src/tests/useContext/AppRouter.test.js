import { AppRouter } from "../../components/useContext/AppRouter"
import {mount} from 'enzyme';
import '@testing-library/jest-dom';
import { UserContext } from "../../components/useContext/context/UserContext";


describe('AppRouter Test', () => {
  test('should render ok', () => {
    
    const wrapper = mount(
    
      <UserContext.Provider value={{user:{name:"nelson", email:"test@test.com"}, setUser: jest.fn()}}>
        <AppRouter />
      </UserContext.Provider>
    );
    
    expect(wrapper).toMatchSnapshot();
  })
  
})
