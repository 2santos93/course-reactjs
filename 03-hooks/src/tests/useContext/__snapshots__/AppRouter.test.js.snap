// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter Test should render ok 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  Array [
    <h1>
      Home
    </h1>,
    <hr />,
    "{
   \\"name\\": \\"nelson\\",
   \\"email\\": \\"test@test.com\\"
}",
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Set User
    </button>,
  ]
</div>
`;
